lib/pavilion/builder.py

```diff
@@ -352,15 +352,24 @@ def _update_src(self):
                 "Source location '{}' points to something unusable."
                 .format(found_src_path))
 
-    def build(self, cancel_event=None):
+    def build(self, cancel_event=None, fail_event=None):
         """Perform the build if needed, do a soft-link copy of the build
         directory into our test directory, and note that we've used the given
         build.
-        :param threading.Event cancel_event: Allows builds to tell each other
-        to die.
+        :param threading.Event cancel_event: Signals build to stop.
+        :param threading.Event fail_event: Allows build to report failure,
+        without cancelling other builds.
         :return: True if these steps completed successfully.
         """
 
+        if fail_event is None:
+            fail_event = threading.Event()
```

Comment:
Do this with the cancel event to. 

Drop checks for None and if it was already set.